skyportal/handlers/api/analysis.py

```diff
@@ -1310,3 +1346,196 @@ async def get(
+                    items:
+                      type: integer
+                    description: |
+                      List of group IDs corresponding to which groups should be
+                      able to view analysis results. Defaults to all of requesting user's
+                      groups.
+        responses:
+          200:
+            content:
+              application/json:
+                schema:
+                  allOf:
+                    - $ref: '#/components/schemas/Success'
+                    - type: object
+                      properties:
+                        data:
+                          type: object
+                          properties:
+                            analysis_id:
+                              type: integer
+                              description: New analysis ID
+        """
+        # allowable resources now are [obj]. Can be extended in the future.
+        if analysis_resource_type.lower() not in ['obj']:
+            return self.error("Invalid analysis resource type", status=403)
+
+        try:
+            data = self.get_json()
+        except Exception as e:
+            return self.error(f'Error parsing JSON: {e}')
+
+        with self.Session() as session:
+
+            stmt = AnalysisService.select(self.current_user).where(
+                AnalysisService.id == analysis_service_id
+            )
+            analysis_service = session.scalars(stmt).first()
+            if analysis_service is None:
+                return self.error(
+                    message=f'Could not access Analysis Service ID: {analysis_service_id}.',
+                    status=403,
+                )
+            if not analysis_service.upload_only:
+                return self.error(
+                    message=f'Analysis Service ID: {analysis_service_id} is not of type upload_only.',
+                    status=403,
+                )
+
+            group_ids = data.pop('group_ids', None)
+            if not group_ids:
+                group_ids = [g.id for g in self.current_user.accessible_groups]
+
+            groups = session.scalars(
+                Group.select(self.current_user).where(Group.id.in_(group_ids))
+            ).all()
+            if {g.id for g in groups} != set(group_ids):
+                return self.error(
+                    f'Cannot find one or more groups with IDs: {group_ids}.'
+                )
+
+            data["groups"] = groups
+            author = self.associated_user_object
+            data["author"] = author
+
+            status_message = data.get("message", "")
+
+            if analysis_resource_type.lower() == 'obj':
+                obj_id = resource_id
+                stmt = Obj.select(self.current_user).where(Obj.id == obj_id)
+                obj = session.scalars(stmt).first()
+                if obj is None:
+                    return self.error(f'Obj {obj_id} not found', status=404)
+                data["obj_id"] = obj_id
```

Comment:
Why are both of these needed?

Reply:
was a cut and paste of legacy code...cleaned it up (not needed)